doc/images.md

```diff
@@ -13,29 +13,113 @@ flutter:
 
 It has to be a PNG file and it can have transparency.
 
+## Loading images
+
+Flame bundles an utility class called `Images` that allows you to easily load and cache images from the assets into the memory.
+
+Flutter has a collection of types related to images, and converting everything properly from a local asset to the Image that can be drawn on Canvas is a bit of a pain. This class allows you to obtain an Image that can be drawn on a Canvas using the `drawImageRect` method.
+
+It automatically caches any image loaded by filename, so you can safely call it many times.
+
+The methods for loading and clearing the cache are: `load`, `loadAll`, `clear` and `clearAll`. They return a `Future` for the Image loaded.
+
+To retrieve a previously cached image, the `fromCache` method can be used.
+
+### Manual use
+
+It can be manually used by instantiating it, like:
+
+```dart
+    import 'package:flame/images.dart';
+    final imagesLoader = Images();
+    Image image = await imagesLoader.load('asd');
+```
+
+But flame also offers to ways of using this class wihtout instantiating it yourself.
+
+### Flame.images
+
+Simply singleton, provided byt the `Flame` class, is can be used as a global image caching.
+
+Example:
+
+```dart
+    import 'package:flame/flame.dart';
+
+    // inside an async context
+    Image image = await Flame.images.load('player.png');
+
+    final playerSprite = Sprite(image);
+```
+
+# Game.images
+
+The `Game` class offers some utility for handling images loading too, it bundles an instance of the `Images` class, that can be used to load images assets that will be used during the game. The game will automatically free the cache when the game widget is removed from the widget tree.
+
+The method `onLoad` from the `Game` class is a great place for the initial assets to be loaded.
+
+Example:
+
+```dart
+class MyGame extends Game {
```

Comment:
BaseGame

Reply:
Not really, I didn't used `BaseGame` on this example snippet.